### DISCUSSION TITLE
ContentDialog customization

### DISCUSSION THREAD

**Discussion by dotNET-anykey** [amwx/FluentAvalonia#359] [Q&A]
Hi, I want to use FluentAvalonia contnent dialog coz of it's mechanics, as it seems the best across market. But the wayt it looks does not fit my application, so I am wondering how to change the appearance of it.
Thanks :)

**Comment by amwx** [comment]
Like all of the controls here, ContentDialog can be customized with a custom ControlTheme. You can use the [normal ControlTheme](https://github.com/amwx/FluentAvalonia/blob/master/src/FluentAvalonia/Styling/ControlThemes/FAControls/ContentDialogStyles.axaml) to see how its put together by default and go from there. The only things that MUST be in the template are the three buttons (as indicated by the `TemplatePart` attribute on the C# class.

**Reply by timunie** [reply]
But I assume the Buttons can still be hidden via IsVisible=false if that is needed.

**Reply by amwx** [reply]
> But I assume the Buttons can still be hidden via IsVisible=false if that is needed.

Yes, but using custom buttons will require you to attach your own handlers and call `Hide(ContentDialogResult)`. The 3 built in buttons automatically handle setting the result and closing the dialog. You can of course restyle them to look however you want.
